app/javascript/packs/addressfinder.js

```diff
@@ -9,7 +9,7 @@
     Array.prototype.forEach.call(elements, function (addressElement) {
       widget = new AddressFinder.Widget(
         addressElement,
-        '<%= Rails.configuration.x.addressfinder.public_key %>',
+        "PKWQ7XJH3GBUCLFMV49Y",
```

Comment:
Why did this change? 🤔

Reply:
> Why did this change?

The previous implementations compiles the _public key_ into the final js (aka, the javascript that the end user gets).

If you go now to production you can see it there:

![image](https://user-images.githubusercontent.com/53300/125743483-dc44a992-d211-46e9-8e79-0d509f77474d.png)

Webpacker (rubygem, not webpack) does not have initial support to compile _erb_.

Although it is possible to add support to compile `.js.erb` I'd rather keep it simple. Because it does not matter how we store it, **anyone** can see it anyway.

Reply:
Is there a way we can avoid the duplication?

You _might_ be able to keep the ERB support. According to [Rails' Webpacker guide](https://edgeguides.rubyonrails.org/webpacker.html), you can run `bin/rails webpacker:install:erb` to get “ERB support on your JavaScript files”.

Reply:
> Is there a way we can avoid the duplication?

There is no duplication. This key is used only here, and it is passed on so the end user can see it.

The other file with this key was `config/initializers/addressfinder.rb` and I had it removed.

> You _might_ be able to keep the ERB support. According to [Rails' Webpacker guide](https://edgeguides.rubyonrails.org/webpacker.html), you can run `bin/rails webpacker:install:erb` to get “ERB support on your JavaScript files”.

Yes, I know. I actually started doing this but I undid… This is a tiny little app, my mindset was: the less dependency/code/complecity the better.